vm.go

```diff
@@ -743,27 +702,48 @@ func (vm *VirtualMachineSpec) AssignDevice(ctx context.Context, deviceID string)
 
 // Removes a vpci device from VirtualMachineSpec
 func (vm *VirtualMachineSpec) RemoveDevice(ctx context.Context, vmBusGUID string) error {
+	return vm.modifySetting(ctx, &hcsschema.ModifySettingRequest{
+		ResourcePath: fmt.Sprintf("VirtualMachine/Devices/VirtualPci/%s", vmBusGUID),
+		RequestType:  requesttype.Remove,
+	})
+}
+
+func (vm *VirtualMachineSpec) GetState() (state string, stopped bool, err error) {
 	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Minute)
 	defer cancel()
 
 	system, err := hcs.OpenComputeSystem(ctx, vm.ID)
 	if err != nil {
-		return err
+		stopped = true
+		return
 	}
-
 	defer system.Close()
 
-	return vm.modifySetting(ctx, &hcsschema.ModifySettingRequest{
-		ResourcePath: fmt.Sprintf("VirtualMachine/Devices/VirtualPci/%s", vmBusGUID),
-		RequestType:  requesttype.Remove,
-	})
+	properties, err := system.Properties(ctx)
+	if err != nil {
+		stopped = true
+		return
+	}
+	return properties.State, properties.Stopped, nil
 }
 
-func (vm *VirtualMachineSpec) OS() string {
+func (vm *VirtualMachineSpec) OS() (string, error) {
 	if vm.gc == nil {
-		return vm.system.OS()
```

Comment:
what's the motivation for this change? why revert the changes?

Reply:
I had put the reason in comment

 

        // The properties of hcs.System are set when hcs.System is created and never refresh
        // cannot use vm.system as vm.system was created at vm.Create() and doesn't have up-to-date properties
        // get the properties from a fresh hcs.System